applications/ShapeOptimizationApplication/python_scripts/mesh_controller_with_solver.py

```diff
@@ -153,4 +188,34 @@ def __FixWholeSurface(optimization_model_part, mesh_solver_settings):
         KM.Logger.PrintInfo("ShapeOpt", "Add automatic process to fix the whole surface to mesh motion solver:")
         mesh_solver_settings["processes"]["boundary_conditions_process_list"].Append(auto_process_settings)
 
+    # --------------------------------------------------------------------------
+    def __CheckAndSetAutomaticMeshRefinementSettings(self, parameters):
```

Comment:
If i understand correctly, the remeshing will also work without any userdefined input?

Or is it necessary to define what metric is used for the remeshing? If yes, where?

Reply:
> If i understand correctly, the remeshing will also work without any userdefined input?
No, it will need user to define an output. But we can provide some defaults for refinement based on geoemtrical properties. What do you suggest?

Only danger in here is, if he uses the fixing process coming from `__FixWholeSurface`, which creates a new `ModelPart` named `auto_surface_nodes` for boundaries in the `Initialize`. This cannot be used with remeshing since this boundary needs an update as well.

Since `MeshMovingAnalysis` allows `boundary_conditions_process_list`, I was using there to manually fix `MESH_DISPLACEMENT` dofs. What do you suggest?
> Or is it necessary to define what metric is used for the remeshing? If yes, where?
Not if you are not using geometrical properties such as "Ratio", but I left the oppportunity to user to decide which to use. All of the options in `MmgProcess` is usable by the user except for the ones I specifically check in here. So he can decide which metric to use.

Reply:
> But we can provide some defaults for refinement based on geometrical properties.

If we can define some defaults for the most basic runable setup, i would try to do that. It would be cool if the most basic setup can be switched on by simply setting `"use_automatic_remeshing": true`

> `auto_surface_nodes`

Is this model_part not automatically filled with nodes after remeshing? If not, we should simply recompute it after the remeshing.

Reply:
Hi! Unresolving this converstaion to discuss the settings a bit more.

As of now, the default settings on the mmg_process.py are using the level-set strategy:

https://github.com/KratosMultiphysics/Kratos/blob/e66d9cc2134ec89c6bbac04ac4e303b69fb9bbf5/applications/MeshingApplication/python_scripts/mmg_process.py#L57

This means that the metric is computed using the DISTANCE function defined by default. If you are not using any DISTANCE value, it would create the metric using all 0, which would assign the metric according to the minimal size given (right now you are using the automatic_remesh option which computes the minimal_size according to the current NODAL_H's). This is probably unwanted by you, unless you are using an embedded approach or a boundary layer where you already have the DISTANCE value.

In any case, I understand that by default you would like to preserve the mesh size constat, only improving mesh quality. There is a built-in function to do that in MMG, which was not supported yet in our interface but Im adding it in #8469.

This will not use any metric field, and will try to preserve the average edge lenghts on the nodes. You can then assign the `strategy : "Optimization"` here and use that as default. If a user wants to use the hessian approach or distance approach, he can override the strategy setting to refine according to that.

This "optimizaton" mode is something that we could do "by hand" as well, assigning a metric value at every node that is equal (or some factor) to the current NODAL_H. However, if the builtin function works fine, I would use that as default.

Reply:
Thanks @marcnunezc for the help and for adding the new interface!

@sunethwarna so i suggest we wait for #8469 to be merged, and i will update the test case using the new remeshing mode.